bundles/org.openhab.binding.publictransportswitzerland/src/main/java/org/openhab/binding/publictransportswitzerland/internal/stationboard/PublicTransportSwitzerlandStationboardHandler.java

```diff
@@ -0,0 +1,144 @@
+        if (command instanceof RefreshType) {
+            // TODO?
+        }
+    }
+
+    @Override
+    public void initialize() {
+        updateStatus(ThingStatus.UNKNOWN);
+
+        //noinspection ConstantConditions
+        if (updateDataJob == null || updateDataJob.isCancelled()) {
+            updateDataJob = scheduler.scheduleWithFixedDelay(this::updateData, 0, 60, TimeUnit.SECONDS);
+        }
+    }
+
+    public void updateData() {
+        PublicTransportSwitzerlandStationboardConfiguration config = getConfigAs(PublicTransportSwitzerlandStationboardConfiguration.class);
+
+        try {
+            String escapedStation = URLEncoder.encode(config.station, StandardCharsets.UTF_8.name());
+            String requestUrl = BASE_URL + "stationboard?station=" + escapedStation + fieldFilters;
+
+            String response = HttpUtil.executeUrl("GET", requestUrl, 10_000);
+            JsonElement jsonObject = new JsonParser().parse(response);
+
+            updateCsvChannel(jsonObject);
+            updateJsonChannel(jsonObject);
+            updateStatus(ThingStatus.ONLINE);
+        } catch (Exception e) {
+            logger.warn("Unable to fetch stationboard data", e);
+
+            updateStatus(ThingStatus.OFFLINE);
+            updateState(CHANNEL_CSV, new StringType("No data available"));
+            updateState(CHANNEL_JSON, new StringType("{}"));
+        }
+    }
+
+    private String createFilterForFields(String... fields) {
+        return Arrays.stream(fields).map((field) -> "&fields[]=" + field).collect(Collectors.joining());
+    }
+
+    private void updateJsonChannel(JsonElement jsonObject) {
+        updateState(CHANNEL_JSON, new StringType(jsonObject.toString()));
+    }
+
+    private void updateCsvChannel(JsonElement jsonObject) throws Exception {
+        JsonArray stationboard = jsonObject.getAsJsonObject().get("stationboard").getAsJsonArray();
+
+        List<String> departures = new ArrayList<>();
+
+        for (JsonElement jsonElement : stationboard) {
+            JsonObject departureObject = jsonElement.getAsJsonObject();
+            JsonObject stopObject = departureObject.get("stop").getAsJsonObject();
+
+            String category = departureObject.get("category").getAsString();
+            String number = departureObject.get("number").getAsString();
+            String destination = departureObject.get("to").getAsString();
+
+            JsonElement delayElement = departureObject.get("delay");
+            String delay = "";
+            if (delayElement != null) {
+                delay = delayElement.getAsString();
+            }
+
+            String departureTime = stopObject.get("departureTimestamp").getAsString();
+            String track = stopObject.get("platform").getAsString();
+
+            departures.add(String.join("\t", category, number, departureTime, destination, track, delay));
+        }
+
+        updateState(CHANNEL_CSV, new StringType(String.join("\n", departures)));
```

Comment:
A multi-lines value is probably not ideal for current OH UIs. I am not sure this is correctly supported.
You could have created 5 channels, one for each departure: departure1, departure2, departure3, departure4 and departure5.

Reply:
Actually I did this to use it in the HabPanel more easily and configure the number of rows more dynamically. (See https://github.com/StefanieJaeger/HABPanel-departure-board/blob/master/sbb-table.widget.json)

Of course I can add the next 5 departures as separate channels, however I would appreciate to be able to keep the CSV channel for the HabPanel module.

Reply:
Can you show with a snapshot the rendering of the csv channel in Basic UI? 
We cannot imagine a binding conceived to work only with a particular UI like HABpanel. 
My proposal is not ideal, it is more a workaround because we have no widget to display a list of strings. 
You can keep the csv channel as an advanced channel.

Reply:
But if the rendering of the channel is ok in all usual apps, you can avoid introducing new channels.